mcs/class/corlib/System/MulticastDelegate.cs

```diff
@@ -112,167 +109,141 @@ public sealed override int GetHashCode ()
 		// </summary>
 		public sealed override Delegate[] GetInvocationList ()
 		{
-			MulticastDelegate d;
-			d = (MulticastDelegate) this.Clone ();
-			for (d.kpm_next = null; d.prev != null; d = d.prev)
-				d.prev.kpm_next = d;
-
-			if (d.kpm_next == null) {
-				MulticastDelegate other = (MulticastDelegate) d.Clone ();
-				other.prev = null;
-				other.kpm_next = null;				
-				return new Delegate [1] { other };
-			}
-
-			var list = new List<Delegate> ();
-			for (; d != null; d = d.kpm_next) {
-				MulticastDelegate other = (MulticastDelegate) d.Clone ();
-				other.prev = null;
-				other.kpm_next = null;
-				list.Add (other);
-			}
-
-			return list.ToArray ();
+			return delegates;
```

Comment:
This is wrong, you must do a defensive copy of the array. Otherwise the caller can change it and break how the internals work.